pandas/tests/frame/test_reshape.py

```diff
@@ -939,3 +939,34 @@ def test_unstack_fill_frame_object():
         index=list('xyz')
     )
     assert_frame_equal(result, expected)
+
+
+def test_transpose_dt64tz():
```

Comment:
Could you also test transposing a mixed dtype dataframe if we aren't already (e.g. datetimetz and int)?

Reply:
Will do.  Only real difference is that it doesn't round-trip (though I think that behavior is correct)

Reply:
Yeah that makes sense. Just good to ensure that there's no aggressive coercion of other data dtypes in the DataFrame once transposed i.e. the columns in `df.T` end up as object.